include/clang/Frontend/CompilerInstance.h

```diff
@@ -183,6 +183,12 @@ class CompilerInstance : public ModuleLoader {
   /// The list of active output files.
   std::list<OutputFile> OutputFiles;
 
+  /// \brief An optional callback function used to wrap all FrontendActions
+  /// produced to generate imported modules before they are executed.
+  std::function<std::unique_ptr<FrontendAction>
+    (FrontendOptions &opts, std::unique_ptr<FrontendAction> action)>
```

Comment:
Change 'FrontendOptions &opts' to 'const FrontendOptions &opts', to disallow the callee from changing the options.